x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/core/type/DataType.java

```diff
@@ -44,7 +44,22 @@
  * processing pipeline, and types which the language doesn't support, but require
  * special handling anyway (e.g. {@link DataType#OBJECT})
  *
+ * <h2>Behavior of new, previously unsupported data types</h2>
+ *
+ * Data types that have support in ES indices, but are not yet supported in ES|QL, are
+ * treated as {@link #UNSUPPORTED} by ES|QL. Fields of that type are filled with
+ * {@code null} values, and no functions support them.
+ * In query plans, these fields amount to
+ * {@link org.elasticsearch.xpack.esql.expression.function.UnsupportedAttribute}s.
+ * <p>
+ * When such a type gets support in ES|QL, query plans cannot contain it
+ * unless all nodes in the cluster (and remote clusters) support it to avoid
```

Comment:
Re `and remote clusters`, not sure if it is worth clarifying, but I believe it is all nodes of remote clusters that are participating in the query, not all connected remote clusters.